pulsar-broker/src/main/java/org/apache/pulsar/PulsarBrokerStarter.java

```diff
@@ -166,6 +166,12 @@ private static boolean argsContains(String[] args, String arg) {
                     "c-" + brokerConfig.getClusterName()
                         + "-fw-" + hostname
                         + "-" + workerConfig.getWorkerPort());
+                // inherit broker authorization setting
+                workerConfig.setAuthorizationEnabled(brokerConfig.isAuthorizationEnabled());
+                workerConfig.setAuthorizationProvider(brokerConfig.getAuthorizationProvider());
+
+                workerConfig.setZooKeeperSessionTimeoutMillis(brokerConfig.getZooKeeperSessionTimeoutMillis());
```

Comment:
don't we need to inherit `configurationStoreServers` from broker?

If we need inherit the settings, it seems to me we are missing a test case here.

Reply:
@sijie the only reason we need the setting ```configurationStoreServers``` is for the current default implementation of ```AuthorizationProvider```, ```PulsarAuthorizationProvider``` which stores all polices/ACLs in zookeeper.  When running the worker service with the broker, the AuthorizationProvider will be pass in to the worker service via PulsarService. Thus, whether ```configurationStoreServers``` is set here or not is irrelevant.  The config is only relevant when running the function worker separately from the broker.  In that scenario, the function worker will need to be able to contact zookeeper.

Reply:
@jerrypeng if we don't set `configurationStoreServers` here, why do we inherit the zookeeper session settings? based on your explanation, it should be either setting all 3 zk settings or don't setting all 3 zk settings. I am confused why we are setting some and not setting the others.

Also if these settings are only relevant when function worker runs separately, you should update the comments in the `function_worker.yml`.

Reply:
@sijie after more carefully looking at the code ```zooKeeperOperationTimeoutSeconds``` is needed here:
https://github.com/apache/pulsar/pull/3874/files#diff-d906ae3d0366fbf1866c0334f075b32eR1943.  This was added recently so that zk operations don't block forever.  The configs ```configurationStoreServers``` and ```zooKeeperSessionTimeoutMillis``` don't need to be inherited by the function worker when running with the broker.  However, for the sake of consistency why don't we just inherit these configs anyways? Maybe in the future the worker will need them in some way.  @sijie how does that sound?

Reply:
sure. that's was my point, we should keep things consistent.